keps/sig-auth/4412-projected-service-account-tokens-for-kubelet-image-credential-providers/README.md

```diff
@@ -0,0 +1,1107 @@
++       // configuration.
++       ServiceAccountAnnotations map[string]string
+ }
+```
+
+Example Kubernetes Service Account:
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: my-service-account
+  namespace: my-namespace
+  annotations:
+    domain.io/identity-id: "12345"
+    domain.io/identity-type: "user"
+    # this annotation will not be passed to the plugin because it's not in tokenAttributes.serviceAccountAnnotationKeys
+    # in the credential provider configuration
+    domain.io/annotation-that-will-not-be-passed: "value"
+```
+
+Example credential provider request:
+
+```json
+{
+  "image": "my-image",
+  "serviceAccountToken": "<service-account-token>",
+  "serviceAccountAnnotations": {
+    "domain.io/identity-id": "12345",
+    "domain.io/identity-type": "user"
+  }
+}
+```
+
+#### Caching Credentials
+
+The cache key generation will be updated to support new caching modes while preserving the existing behavior when the audience field is not set.
+
+##### Existing behavior
+
+If the `serviceAccountTokenAudience` field is not set in the provider configuration for the plugin, the key generation should fall back to the current behavior: using only the image registry URL or other existing identifiers.
+
+##### New behavior when the `serviceAccountTokenAudience` field is set
+
+Based on the `PluginCacheKeyType` definitions (Image, Registry, Global), here’s a breakdown of how cache keys would be generated:
+
+1. `GlobalPluginCacheKeyType` (Global)
+   - Global cache key type will not be allowed to use with the `tokenAttributes.serviceAccountTokenAudience` field configured for the plugin.
+2. `RegistryPluginCacheKeyType` (Registry)
+   - The cache key will be generated based on the image registry URL, service account metadata (namespace, name, UID), and hash of the service account annotations that are passed to the plugin in the `CredentialProviderRequest.ServiceAccountAnnotations` field.
+3. `ImagePluginCacheKeyType` (Image)
+    - The cache key will be generated based on the image URL, service account metadata (namespace, name, UID), and hash of the service account annotations that are passed to the plugin in the `CredentialProviderRequest.ServiceAccountAnnotations` field.
+
+#### How will this work with the Ensure secret pull images KEP?
+
+[2535-ensure-secret-pull-images](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2535-ensure-secret-pulled-images) is a KEP that aims to give admin the ability to ensure pods that use a image are authorized to access the image. The KEP doesn't factor in the use of service account tokens for image pull. As part of the alpha implementation the 2 features together will not provide a desired outcome. Post alpha, we'll need to update the Ensure secret pull images KEP to factor in the use of service account tokens for image pull.
+
+Notes from reviewing the current KEP and discussion with @stalz:
```

Comment:
stalz would've been a much-easier-to-pronounce handle now that I see it :slightly_smiling_face: